### PR TITLE
[WIP] Issue #1909 Adding GUI display when saving logic is  triggered 

### PR DESCRIPTION
**Pull request by Taguiar**
@Pomax and @humphd 
I finished with the initial logic for GUI display whenever the user is logged out and his modifications are saved to the browser.
In terms of the design, I tried to play safe since and used whatever is already been display on the case that the user is logged on. I much rather have @flukeout giving me his thoughts about the design before I spend time in something not desirable.

**When modifications are been made:**
untitled

**When modifications have been successfully saved:**
capture1

### CODE REVIEW

**Comment by Taguiar** [comment]
Ok I will see what I can do!

**Comment by flukeout** [comment]
Testing this out now.

**Comment by flukeout** [comment]
Hey OK, so is this PR only meant to address the Signed Out case when we save the files to the browser but not "to the cloud"? If so, I think there are just two states to worry about...

* Changes detected, saving begins...
  * Indicator says "Saving..."
* Changes saved to locally to browser only
  * Indicator says "Saved in browser" <- this stays until the previous case is detected again.

However, this doesn't seem all that important because it basically always saves to the browser. The more important case would be to address the "All files saved to Cloud" scenario mentioned by @humphd above. 

Will you address that case here @Taguiar ?

**Comment by humphd** [comment]
I think the not-signed-in case is still really important, @flukeout, since it's not common for me to go to a site and assume that whatever I just spent an hour typing will automatically be saved when I refresh or close the tab--we need to provide some assurance.

I like your idea of always having "Saved in browser" and then changing it to "Saving..." when that is happening.  @Taguiar please evolve your code to do that, and to do it in ways that work with what I mentioned in my review.

RE: "All files saved to the Cloud" is another bug we can do on top of, and in addition to this, and it will happen in the signed-in case.

**Comment by flukeout** [comment]
That sounds like what we need @humphd

**Comment by Taguiar** [comment]
It is working a little bit more as you guys expected @humphd and @flukeout 
Although the internal communication is a bit hard to detect even me doing exactly the same thing as previous implementations!
I was thinking to follow this instructions here. But I would like some guidance before I do something very wrong!

**Comment by humphd** [comment]
@gideonthomas is the localization master, I'd ask him for help.

**Comment by gideonthomas** [comment]
@Taguiar I'm not sure I understand your question. What you have in your PR is correct. The steps you have linked to is the right way to generate the localized strings. You might just need to restart the server with `npm run restart-server`

**Comment by flukeout** [comment]
Within thimble, to get strings running this also works. In the terminal form your thimble folder...

* vagrant ssh
* cd /vagrant
* npm run localize

File: locales/en-US/messages.properties [hunk 1]
```diff
@@ -42,6 +42,7 @@ renameProjectSaveBtn=Save
 renameProjectSavingIndicator=Saving...
 fileSavingIndicator=Saving...
 fileSavingFailedIndicator=Saving failed, retrying...
+dirtyFileSaved=Saved in Browser.
```
Comment (humphd) [inline locales/en-US/messages.properties]: `projectFilesSaved=All files saved offline.`

File: views/editor/userbar.html [hunk 2]
```diff
@@ -17,6 +17,10 @@
       <div id="navbar-save-indicator" class="hide">
         {{ gettext("fileSavingIndicator") }}
       </div>
+      <div id="navbar-dirt-file">
```
Comment (humphd) [inline views/editor/userbar.html]: Rename this to `id="navbar-saved-offline-indicator"`

File: public/editor/scripts/editor/js/fc/bramble-ui-bridge.js [hunk 3]
```diff
@@ -443,6 +443,17 @@ define(function(require) {
       }
     });
 
+    bramble.on("projectDirty", function(){
+      $("#navbar-dirt-file").addClass("hide");
+
```
Comment (humphd) [inline public/editor/scripts/editor/js/fc/bramble-ui-bridge.js]: Remove this blank line.

**Comment by Taguiar** [comment]
Changes requested by @humphd.

File: views/editor/userbar.html [hunk 4]
```diff
@@ -17,6 +17,9 @@
       <div id="navbar-save-indicator" class="hide">
         {{ gettext("fileSavingIndicator") }}
       </div>
+      <div id="navbar-saved-offline-indicator">
```
Comment (gideonthomas) [inline views/editor/userbar.html]: you want this `div` only when the user is logged in. So this code needs to wrapped in a nunjucks `if` block similar to https://github.com/mozilla/thimble.mozilla.org/blob/master/views/editor/userbar.html#L101-L106

File: public/editor/scripts/editor/js/fc/bramble-ui-bridge.js [hunk 5]
```diff
@@ -443,6 +443,16 @@ define(function(require) {
       }
     });
 
+    bramble.on("projectDirty", function(){
+      $("#navbar-dirt-file").addClass("hide");
```
Comment (gideonthomas) [inline public/editor/scripts/editor/js/fc/bramble-ui-bridge.js]: You need to do this only for the "not-signed-in" case. I bet that can be done with a `!Project.getUser()` check

**Comment by humphd** [comment]
Closing in favour of https://github.com/mozilla/thimble.mozilla.org/pull/2209, where I'm finishing this.

**Pull request closed without merging**